client-ui/src/app/components/lineage/lineage-details/operation-properties-details/operation-properties-details.component.ts

```diff
@@ -94,18 +88,26 @@ export class OperationPropertiesDetailsComponent implements AfterViewInit, OnDes
     return this.store.select('detailsInfos')
   }
 
-  public getIcon(operationName: string): string {
-    return String.fromCharCode(operationIconCodes.get(operationName) || operationIconCodes.get(OperationType.Generic))
+  public getOperationIcon(operationType: string, operationName: string): string {
+    return getOperationIcon(operationType, operationName)
   }
 
-  public getOperationColor(operationName: string): string {
-    return operationColorCodes.get(operationName) || operationColorCodes.get(OperationType.Generic)
+  public getOperationColor(operationType: string, operationName: string): string {
+    return getOperationColor(operationType, operationName)
   }
```

Comment:
Does this work? At first glance it looks like endless cycle.

Reply:
Ye because it's calling getOperationColor from another utilit file. Let me rename it